### PR TITLE
bs: merge check-build-system and check-types

### PR DESCRIPTION
**Pull request by samouri**
**summary**
Addresses root cause of https://github.com/ampproject/amphtml/pull/37694

1. Combines `check-types` with `check-build-system`.
2. Creates a new "Target" for capturing extensions and causes check-types to trigger on that.

**todo**

Dependency-based target

### CODE REVIEW

File: build-system/tasks/check-types.js [hunk 1]
```diff
@@ -8,13 +8,29 @@ const {compileJison} = require('./compile-jison');
 const {cyan, green} = require('kleur/colors');
 const {execOrThrow} = require('../common/exec');
 const {log} = require('../common/logging');
+const {updateSubpackages} = require('../common/update-packages');
+const fastGlob = require('fast-glob');
```
Comment (rcebulko) [inline build-system/tasks/check-types.js:12]: don't we use globby in most places?

File: build-system/tasks/check-types.js [hunk 2]
```diff
@@ -8,13 +8,29 @@ const {compileJison} = require('./compile-jison');
...
+const path = require('path');
+
+/**
+ * Helper that updates build-system subpackages so their types can be verified.
+ * Skips npm checks during CI (already done while running each task).
+ */
+function updateBuildSystemSubpackages() {
+  const packageFiles = fastGlob.sync('build-system/tasks/*/package.json');
```
Comment (rcebulko) [inline build-system/tasks/check-types.js:20]: Do we want just `/*/` or `/**/` here? ie. should it include subfolders? Guessing not but confirming

**Review by rcebulko — APPROVED** [review]

File: build-system/tasks/check-types.js (hunk 1 above)
Comment (samouri) [reply build-system/tasks/check-types.js:12]: we moved away from it AFAIK b/c it doesn't support windows

**Comment by samouri** [comment]
Moving back to DRAFT as @alanorozco and I have been noodling on a _correct_ way of doing this.

File: build-system/tasks/pr-check.js [hunk 3]
```diff
@@ -93,10 +89,19 @@ async function prCheck() {
...
     runCheck('amp dep-check');
-    runCheck('amp check-types');
     runCheck('amp check-sourcemaps');
   }
 
+  if (
+    buildTargetsInclude(
+      Targets.RUNTIME,
+      Targets.BUILD_SYSTEM,
+      Targets.EXTENSIONS
```
Comment (rsimha) [inline build-system/tasks/pr-check.js:99]: Drive-by comment: This only takes care of the logic for the `amp pr-check` task. We'll also need to update every other spot that includes `Targets.RUNTIME` to also include `Targets.EXTENSIONS`, without which CI will stop running tests for PRs that modify extensions.

**Review by rileyajones — APPROVED** [review]
Thanks for doing this cleanup

File: build-system/tasks/check-types.js (hunk 2 above)
Comment (alanorozco) [reply build-system/tasks/check-types.js:20]: There are `package.json` files deeper so I think we indeed need the recursive wildcard.

```suggestion
  const packageFiles = fastGlob.sync([
    'build-system/tasks/**/package.json',
    '!**/node_modules/**',
  ]);
```

**Pull request closed without merging**